### PR TITLE
tag-#77:  From validation

### PR DESCRIPTION
**Pull request by MrBlackGhostt**
## Summary

Use the React-hook-form and Zod for the form validation

## Description

1.Install Dependencies:
First, make sure you have react-hook-form and zod installed. You can install them using npm or yarn:

2.Create Form Schema with Zod:
Define your form schema using Zod. This includes specifying the validation rules and error messages for each field.

3.Use New Dependencies:
use the React-hook-form & zod for the input and validation

## Images

Before
Screenshot 2023-12-15 at 8 31 08 PM

After

Screenshot 2023-12-15 at 8 31 17 PM

- Template should be strictly **Closes <issue_number>**
- Example: Closes #1

## Prerequisites

- [x] Have you followed all the [CONTRIBUTING GUIDELINES](https://github.com/krishnaacharyaa/wanderlust/blob/main/.github/CONTRIBUTING.md#guidelines-for-contributions)?

### CODE REVIEW

**Comment by krishnaacharyaa** [comment]
@MrBlackGhostt we know the error, kindly check the discord

File: frontend/src/lib/blog.zod.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { z } from 'zod';
+
+export const FormDataSchema = z.object({
+  title: z.string().refine((value) => value.trim().split(/\s+/).length > 3, {
+    message: 'Oops! Title needs more spice. Give it at least 3 words.',
+  }),
+  authorName: z.string().refine((value) => value.trim().length >= 3 && value.trim().length <= 15, {
+    message: 'Hey, Your name should be more than 5 letters.',
+  }),
```
Comment (krishnaacharyaa) [inline frontend/src/lib/blog.zod.ts:9]: We can't use "your name **should** be" emphasis on should, becuase he can be Raj, 
We can be like "C'mon your name cannot be less than 3 characters" ( if its less than 3 letters);
if greater than 15 say "Hey isn't it too big of a name, can you limit it to 15 characters" or improvise on this

File: frontend/src/lib/blog.zod.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  imageLink: z.string().refine((value) => isValidImageLink(value), {
+    message: 'Hmm... Image link should end with .jpg, .jpeg, .webp, or .png.',
+  }),
+  categories: z.array(z.string()).refine((value) => value.length <= 3, {
+    message: 'Easy there! Select up to 3 categories.',
+  }),
```
Comment (krishnaacharyaa) [inline frontend/src/lib/blog.zod.ts:15]: Nice!

**Review by krishnaacharyaa — CHANGES_REQUESTED** [review]
image
Functionalities are not perfectly working.
Title though it is 3 words, The error isn't resolved,

And categories selection is broken, kindly address the comments and this, and also resolve the merge conflicts by rebasing your branch with the latest main branch

File: frontend/src/lib/blog.zod.ts (hunk 2 above)
Comment (MrBlackGhostt) [reply frontend/src/lib/blog.zod.ts:33]: done the changes

**Comment by krishnaacharyaa** [comment]
image
image

Observations.
1. Categories selection is broken, selection is shown only after the button is cilcked
2. If you give everything right at the first go it doesn't let you create post, and doesn't even compliant
3. The dynamic error handling for the images and categories isn't worknig
4. Typo in "C'ome on! Your name cannot be less than 3 characters."
5. There are conflicts with the main branch, kindly rebase.

Request review once all of these is handled, if you feel this is too much of a task, seek help from others in the community discord,
Thank you.

File: frontend/src/lib/blog.zod.ts (hunk 2 above)
Comment (MrBlackGhostt) [reply frontend/src/lib/blog.zod.ts:26]: now I see the Sarcism here 😅

**Comment by krishnaacharyaa** [comment]
If we are not working on this anymore, kindly close this

**Pull request closed without merging**